### PR TITLE
Fix templating multi-document OSD specs

### PR DESCRIPTION
**Pull request by cityofships**

### CODE REVIEW

File: roles/cephadm/README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Role variables
     * `cephadm_osd_spec`: OSD spec to apply in yaml format
       Example:
       ```
-          cephadm_osd_spec:
+          cephadm_osd_spec: |
```
Comment (markgoddard) [inline roles/cephadm/README.md:49]: This breaks backwards compatibility

Comment (cityofships) [reply roles/cephadm/README.md:49]: Indeed. Can be released as 2.0.0. If only we had a changelog...

Comment (markgoddard) [reply roles/cephadm/README.md:49]: Feel free to create one

Comment (cityofships) [reply roles/cephadm/README.md:49]: Yup, on it

Comment (markgoddard) [reply roles/cephadm/README.md:49]: Is there any way to handle both formats?

Comment (cityofships) [reply roles/cephadm/README.md:49]: @mnasiadka suggested one

File: roles/cephadm/tasks/osds.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@
     vars:
       fsid: "{{ cephadm_fsid if cephadm_fsid | length > 0 else cephadm_fsid_current.stdout }}"
     copy:
-      content: "{{ cephadm_osd_spec | to_nice_yaml(indent=2) }}"
+      content: "{{ cephadm_osd_spec | to_nice_yaml if cephadm_osd_spec | type_debug == 'dict' else cephadm_osd_spec }}"
```
Comment (markgoddard) [inline roles/cephadm/tasks/osds.yml:22]: I think you can use `is mapping`

**Review by markgoddard — APPROVED** [review]

**Review by mnasiadka — APPROVED** [review]

**Comment by cityofships** [comment]
Thanks for reviewing 👍

**Pull request merged**